kiwixbuild/builder.py

```diff
@@ -125,15 +125,23 @@ def build(self):
     def _get_packages(self):
         packages_list = []
         for config in ConfigInfo.all_running_configs.values():
+            # get {host}_{config} packages
             mapper_name = "{host}_{config}".format(
                 host=neutralEnv("distname"), config=config
             )
             package_name_mapper = PACKAGE_NAME_MAPPERS.get(mapper_name, {})
             packages_list += package_name_mapper.get("COMMON", [])
+            # get {host}_{codename}_{config} packages
+            mapper_name = "{host}_{codename}_{config}".format(
+                host=neutralEnv("distname"), codename=neutralEnv("codename"), config=config
+            )
+            package_name_mapper = PACKAGE_NAME_MAPPERS.get(mapper_name, {})
+            packages_list += package_name_mapper.get("COMMON", [])
 
```

Comment:
I don't like the code duplication introduced here and in the other change in this (`_get_packages()`) function.

You can avoid it with the help of an auxiliary function `_get_mapper_names_for_config(config)`:

```python
def _get_mapper_names_for_config(config):
    host = neutralEnv("distname")
    codename = neutralEnv("codename")
    return ( f"{host}_{config}", f"{host}_{codename}_{config}" )
```

Then your code here will turn into:

```python
        for config in ConfigInfo.all_running_configs.values():
            for mapper_name in _get_mapper_names_for_config(config):
                package_name_mapper = PACKAGE_NAME_MAPPERS.get(mapper_name, {})
                packages_list += package_name_mapper.get("COMMON", [])
```

Reply:
Understood about the duplication. There aren't any unit tests so I was avoiding any refactors, but I can adjust as you requested. May need a few days before I can get to it.